autogen/agentchat/conversable_agent.py

```diff
@@ -839,7 +839,7 @@ def generate_code_execution_reply(
             return False, None
         if messages is None:
             messages = self._oai_messages[sender]
-        last_n_messages = code_execution_config.pop("last_n_messages", 1)
+        last_n_messages = code_execution_config.pop("last_n_messages", "auto")
```

Comment:
Could use a validation here to make sure the popped string is indeed "auto".

Reply:
Good point. 

Note that the current code is pretty-much identical and lacks validation, so this would strictly be an enhancement.

Reply:
All config dictionaries should be replaced with Pydantic classes with proper validations, but that should be a separate PR that would replace all those cases.

Reply:
Agreed re:Pydantic. Until then, I've added validation.